misc/output_test.py

```diff
@@ -130,6 +130,31 @@ def test_pr_1685(self):
         self.assertEqual(run('', flags='-t recompact'), '')
         self.assertEqual(run('', flags='-t restat'), '')
 
+    def test_pr_2390(self):
+        with tempfile.TemporaryDirectory() as d:
+            os.chdir(d)
```

Comment:
FYI, there is a subtle issue when changing the current directory to the temporary one, as in reality this prevents the temporary directory from being removed on scope exit! Also this breaks pytest when adding other _test.py files in the misc/ directory.

This problem already exists in the run() function above, and I'll submit a patch soon to fix this. Can I ask you to avoid this os.chdir() completely here by doing the following:

1) Use `os.path.join(d, 'build.ninja')` and `os.path.join(d, `.ninja_log`)` when writing these files below.
2) Use `cwd=d` when invoking subprocess.check_output()? This works because `NINJA_PATH` is always an absolute path (see line 21 above).